docs/diplomat-and-macros.md

```diff
@@ -0,0 +1,186 @@
+# Comparing UniFFI with Diplomat
+
+[Diplomat](https://github.com/rust-diplomat/diplomat/) and [UniFFI](https://github.com/mozilla/uniffi-rs/)
+are both tools which expose a rust implemented API over an FFI.
+At face value, these tools are solving the exact same problem, but their approach
+is significantly different.
+
+This document attempts to describe these different approaches and discuss the pros and cons of each.
+It's not going to try and declare one better than the other, but instead just note how they differ.
+If you are reading this hoping to find an answer to "what one should I use?", then that's easy -
+each tool currently supports a unique set of foreign language bindings, so the tool you should
+use is the one that supports the languages you care about!
+
+(There may even be a future where these 2 tools converge - that seems like a lot of work, but
+might also provide a large payoff - more on this later)
+
```

Comment:
i think a difference not really covered in this document is the style of actual interface. UniFFI basically seems to be serializing types across FFI with this cool FFIConverter scheme, whereas Diplomat uses raw `repr(C)` so backends do not need much ceremony to read things. Both are valid approaches, but it's an interesting difference likely borne out of the choice of languages to prioritize (Diplomat cares heavily about C++)

Reply:
As Ryan notes later, I think this is a reflection of expedience and initial use-cases, but not necessarily a key difference between the tools at a high level - I think UniFFI *could* support "native" `repr(C)` types in some cases.

If you still think this is important to communicate after seeing my updated version, please give me a rough idea of what it should say and where :)